include/worker.hpp

```diff
@@ -100,6 +100,46 @@ class Worker
                       types::ObjectMap& objectInterfaceMap,
                       const std::string& vpdFilePath);
 
+    /**
+     * @brief An API to get backup and restore path information.
+     *
+     * This API is to get backup and restore's source1 or source2, EEPROM or
+     * inventory path information present in the config JSON, based on the input
+     * i_pathName.
+     *
+     * @param[in] i_pathName - source1 or source2.
+     * @param[out] o_location - Holds hardware or inventory string value, for
+     * given i_pathName.
+     * @param[out] o_path - Holds EEPROM or inventory path, for given
+     * i_pathName.
+     */
+    void getBackupAndRestorePathInfo(const std::string& i_pathName,
+                                     std::string& o_location,
+                                     std::string& o_path);
+
+    /**
+     * @brief An API to backup and restore VPD.
+     *
+     * Note: This API works on the set of pre-declared keywords.
+     * Restore or backup action could be triggered for each keyword,
+     * based on the keyvalue present in the source1 and source2
+     * keyword.
+     *
+     * Restore source1 keyword value with source2 keyword value,
+     * when source1 keyword has default value but
+     * source2 has non default value.
+     *
+     * Backup the source1 keyword value on source2,
+     * when source1 keyword has non default value but
+     * source2 has default value.
+     *
+     * @param[in,out] io_source1VpdMap - Source1 VPD map.
+     */
+    void backupAndRestore(types::VPDMapVariant& io_source1VpdMap);
+
+    // To perform backup and restore
+    bool performBackupAndRestore;
```

Comment:
Make it private with constant getter. No need of having setter.

Reply:
Done.